core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -280,13 +285,18 @@ public RemoteStorageManager storageManager() {
         return remoteLogStorageManager;
     }
 
+    public MetadataCache metadataCache() {
+        return metaDataCache;
+    }
+
     private Stream<Partition> filterPartitions(Set<Partition> partitions) {
         // We are not specifically checking for internal topics etc here as `log.remoteLogEnabled()` already handles that.
         return partitions.stream().filter(partition -> partition.log().exists(UnifiedLog::remoteLogEnabled));
     }
 
     private void cacheTopicPartitionIds(TopicIdPartition topicIdPartition) {
-        Uuid previousTopicId = topicPartitionIds.put(topicIdPartition.topicPartition(), topicIdPartition.topicId());
+        Map<String, Uuid> mapping = metaDataCache.topicNamesToIds();
```

Comment:
Adding to Divij's comment - this is an unmodifiable map. Even if you wanted to you wouldn't be able to add to it:
```
  def topicNamesToIds(): util.Map[String, Uuid] = {
    Collections.unmodifiableMap(metadataSnapshot.topicIds.asJava)
  }
```

Reply:
Thanks. I've removed this logic